gorb/gorb.go

```diff
@@ -0,0 +1,299 @@
+		}
+	}
+
+	log.Infof("Waiting %v to finish gorb draining", g.drainDelay)
+	time.Sleep(g.drainDelay)
+
+	err := g.ManageLoopBack("del", 0, 0)
+	errorArr = multierror.Append(errorArr, err)
+
+	for _, backend := range g.backend {
+		err := g.removeBackend(&backend)
+		if err != nil {
+			log.WithFields(log.Fields{"err": err}).Error("Unable to remove Backend ")
+			errorArr = multierror.Append(errorArr, err)
+		} else {
+			log.WithFields(log.Fields{"backend": backend}).Infof("Backend succesfully removed")
+		}
+	}
+
+	return errorArr.ErrorOrNil()
+}
+
+func (g *gorb) Health() error {
+	resp, err := g.httpClient.Get(fmt.Sprintf("%s/service", g.serverBaseURL))
+	if err != nil {
+		return fmt.Errorf("Unable to check service details: %v", err)
+	}
+
+	if resp.StatusCode != 200 {
+		return fmt.Errorf("Gorb server is not healthy. Status code: %d", resp.StatusCode)
+	}
+
+	return nil
+}
+
+func (g *gorb) Update(controller.IngressEntries) error {
+
+	var errorArr *multierror.Error
+	err := g.ManageLoopBack("add", 1, 2)
+	errorArr = multierror.Append(errorArr, err)
+
+	for _, backend := range g.backend {
+
+		backendExists, err := g.backendExists(&backend)
+		if err != nil {
+			log.WithFields(log.Fields{"err": err}).Error("Unable to check if backend already exists")
+		}
+
+		if backendExists == 404 {
+			err := g.addBackend(&backend)
+			if err != nil {
+				log.WithFields(log.Fields{"err": err}).Error("Error add Backend ")
+				errorArr = multierror.Append(errorArr, err)
+			} else {
+				log.WithFields(log.Fields{"backend": backend}).Infof("Backend added successfully")
+			}
+		}
+	}
+
+	return errorArr.ErrorOrNil()
+}
+
+func (g *gorb) backendExists(backend *Backend) (int, error) {
+	resp, err := g.httpClient.Get(fmt.Sprintf("%s/service/%s/node-%s-%s", g.serverBaseURL, backend.ServiceName, backend.ServiceName, g.instanceIP))
+	if err != nil {
+		return 0, fmt.Errorf("Unable to retrieve backend details for instance ip: %s error :%v", g.instanceIP, err)
+	}
+	defer resp.Body.Close()
+	return resp.StatusCode, nil
+}
+
+func (g *gorb) addBackend(backend *Backend) error {
+	payload, err := json.Marshal(backend.BackendConf)
+
+	req, err := http.NewRequest("PUT", fmt.Sprintf("%s/service/%s/node-%s-%s", g.serverBaseURL, backend.ServiceName, backend.ServiceName, g.instanceIP), bytes.NewBuffer(payload))
+	if err != nil {
+		return fmt.Errorf("Error while creating add backend request: %v", err)
+	}
+
+	resp, err := g.httpClient.Do(req)
+	if err != nil {
+		return fmt.Errorf("Error while adding backend: %v, error: %v", backend, err)
+	}
+	if resp.StatusCode != 200 {
+		body, _ := ioutil.ReadAll(resp.Body)
+		log.WithFields(log.Fields{"StatusCode": resp.StatusCode}).Infof("Error PUT add backend")
+		return fmt.Errorf("Failed to add backend: %v, status code: %d, response: %v", backend, resp.StatusCode, body)
+	}
+	return nil
+}
+
+func (g *gorb) modifyBackend(backend *Backend) error {
```

Comment:
Would I be right in thinking the difference between this and `addBackend` is just the HTTP method, and the content of a few log messages? If so, I think we could remove the duplicate code.

Reply:
That's true but in the same time, we will need to provide 3 or 4 parameter for customising the log, it will add more complexity on the method. 
I'm happy to discuss about that, I was thinking to do that when I write the code.

Reply:
The refactoring you've done to introduce `executeRequest` looks good to me.